app/models/rbac.rb

```diff
@@ -496,7 +496,7 @@ def self.search(options = {})
       auth_count  = targets.length
     end
 
-    unless search_filter.nil?
+    if search_filter && targets && (!exp_attrs || !exp_attrs[:supported_by_sql])
```

Comment:
@kbrock I double checked how supported_by_sql is being set and this looks good. Should be a pretty good perf improvement. Not sure why it wasn't always checking it.